### ISSUE TITLE
Support absolute URL in `SFLocalHelpers.loadFile()`

### ISSUE THREAD

**Issue by rikmarais** [etriebe/dnd-randomizer#46]
Follows up on #43

Hey there @etriebe ! 👋 
Thanks so much for all the improvements you brought in with #44

The `ensureFolder` call now recognizes when a folder exists in the Forge Assets Library and no longer tries to create new directories every time 🥳

Unfortunately, there's one other issue that I ran into and that is that `SFLocalHelpers.loadFile()` doesn't handle absolute paths very well due to code that replaces the cacheFolder (a relative folder) in the path. 

```js
	  const cacheFolder = SFLocalHelpers.getSystemCacheFolder();
      await this.ensureFolder(cacheFolder);
      if (hasCacheFolderInPath)
      {
        fileName = fileName.replace(cacheFolder, "").replace("/","");
      }

```

This works well when the path is relative, but when the path returned by `FilePicker.browse()` is an absolute URL (starting with `https://` and hosted on a server somewhere, the `cacheFolder` might be in the middle of the absolute URL, so when the `fullFilePath` is reconstructed by inserting the `cacheFolder` at the start of the `fileName` in 
```js
let fullFilePath = this.getCachePath(fileName);
```
you might end up with something like `cacheFolder/https://path.to.json`

I think that a good way around this could be to check if the URL is absolute first and if it is, then try to load it directly.
The browser will cache the fetched file, meaning that performance won't be impacted by subsequent fetches.
And if there is a failure, we can catch it and resort to the relative path check with the module's built-in caching as a fallback.

We're close!

**Comment by rikmarais** [comment]
@etriebe I've added a PR which adds support for absolute URLs to `SFLocalHelpers.loadFile()`. I tested this out on The Forge and it sped up `Generate Encounters` from close to a minute to a few seconds 🥳 

I hope this helps the Forge related performance issues you mentioned before!

By the way, The Forge offers free developer accounts for module developers so that they can test their modules on that system if needed. If you'd like one for testing, swing on by the Forge [Discord](https://forge-vtt.com/discord) and make a ticket in `#ticket-support` mentioning that you're a module dev to get set up :)
